Mathlib/Algebra/Category/GroupCat/Injective.lean

```diff
@@ -152,4 +162,273 @@ instance injective_of_divisible [DivisibleBy A ℤ] :
             congr
 #align AddCommGroup.injective_of_divisible AddCommGroupCat.injective_of_divisible
 
+instance injective_ratCircle : CategoryTheory.Injective <| of <| ULift.{u} <| AddCircle (1 : ℚ) :=
+  have : Fact ((0 : ℚ) < 1) := ⟨by norm_num⟩
+  @injective_of_divisible _ _
+  { div := λ a n ↦ ULift.up (DivisibleBy.div a.down n)
+    div_zero := λ a ↦ by simp only [DivisibleBy.div_zero]; rfl
+    div_cancel := λ {n} a ha ↦ by ext1; exact DivisibleBy.div_cancel a.down ha }
```

Comment:
This looks like a missing `Divisible` instance for `ULift`; can you add it in the appropriate place?

Reply:
I have moved this to `GroupTheory/Divisible.lean`